### PR TITLE
Make test tag filters behave as intended

### PR DESCRIPTION
**Pull request by frreiss**
`configure.py` adds multiple instances of the `--test_tag_filters` and `--build_tag_filters` arguments to the bottom of `.tf_configure.bazelrc`. For example, on a Mac, the configuration file contains the lines:
```
test --test_tag_filters=-benchmark-test,-no_oss,-oss_serial
test --build_tag_filters=-benchmark-test,-no_oss
test --test_tag_filters=-gpu,-nomac,-no_mac
test --build_tag_filters=-gpu,-nomac,-no_mac
```
These lines are intended to disable tests with the tags `benchmark-test`, `no_oss`, `oss_serial`, `gpu`, `nomac`, and `no_mac`. Unfortunately, Bazel's `--test_tag_filters` and `--build_tag_filters` commands replace their respective lists of filters instead of adding to them. So only the last set of filters (`-gpu,-nomac,-no_mac` in this case) ends up getting applied. As a result the default Bazel configuration will run tests that shouldn't run.

This PR modifies `configure.py` so that the script generates a single value for the `--test_tag_filters` and `--build_tag_filters` arguments. After this change, the lines of `.tf_configure.bazelrc` described above turn into:
```
test --test_tag_filters=-benchmark-test,-no_oss,-gpu,-nomac,-no_mac,-oss_serial
test --build_tag_filters=-benchmark-test,-no_oss,-gpu,-nomac,-no_mac
```

### CODE REVIEW

File: configure.py [hunk 1]
```diff
@@ -1155,29 +1155,33 @@ def system_specific_test_config(env):
...
-  write_to_bazelrc('test --build_tag_filters=-benchmark-test,-no_oss')
+
+  # Each instance of --test_tag_filters or --build_tag_filters overrides all
+  # previous instances, so we need to build up a complete list and write a
+  # single list of filters for the .bazelrc file.
+
+  # Filters to use with both --test_tag_filters and --build_tag_filters
+  common_filters = ['-benchmark-test', '-no_oss']
+  # Additional filters for --test_tag_filters beyond those in common_filters
+  addl_test_filters = ['-oss_serial']
```
Comment (mihaimaruseac) [inline configure.py]: I would rename it to `test_only_filters`

File: configure.py [hunk 2]
```diff
@@ -1155,29 +1155,33 @@ def system_specific_test_config(env):
...
-  write_to_bazelrc(
-      'test --test_tag_filters=-benchmark-test,-no_oss,-oss_serial')
-  write_to_bazelrc('test --build_tag_filters=-benchmark-test,-no_oss')
+
+  # Each instance of --test_tag_filters or --build_tag_filters overrides all
+  # previous instances, so we need to build up a complete list and write a
+  # single list of filters for the .bazelrc file.
+
+  # Filters to use with both --test_tag_filters and --build_tag_filters
+  common_filters = ['-benchmark-test', '-no_oss']
```
Comment (mihaimaruseac) [inline configure.py]: Rename these to `test_and_build_filters` please. Gives a clearer meaning to where they apply

**Review by mihaimaruseac — COMMENTED** [review]
Thank you for the PR. I have some suggestions regarding readability.

File: configure.py (hunk 1 above)
Comment (frreiss) [reply configure.py]: Good idea! I've renamed this variable in commit bab7d7d

File: configure.py (hunk 2 above)
Comment (frreiss) [reply configure.py]: Renamed this variable in commit bab7d7d

**Review by mihaimaruseac — APPROVED** [review]

**Pull request merged**